waku/v2/api/filter/filter.go

```diff
@@ -213,12 +214,14 @@ func (apiSub *Sub) multiplex(subs []*subscription.SubscriptionDetails) {
 	for _, subDetails := range subs {
 		apiSub.subs[subDetails.ID] = subDetails
 		go func(subDetails *subscription.SubscriptionDetails) {
+			defer utils.LogOnPanicAndRethrow()
```

Comment:
There are many places in the code where go-routines are not started with `go func(` rather `go <func name>`. 
Maybe good to search with `go<space>` and replace them.
We will have to invoke this in such places by replacing it with a `go func(` style and with defer. 

some examples, but there are a lot more: 
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L85
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L103
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L138
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/api/filter/filter_manager.go#L231
- https://github.com/waku-org/go-waku/blob/798c9c5d819ac8f01cd77912dcd80a702c5ed3b5/waku/v2/node/wakunode2.go#L355